src/JCE.Business/Services/ProfileService.cs

```diff
@@ -56,17 +57,91 @@ public async Task<List<GetProfileDto>> GetUserProfilesByFilter(GetProfileDto fil
 
     }
 
+    public async Task<SuccesfulCreatedProfileDto> CreateUserProfile(CreateProfileDto user)
+    {
+        //Validate that every field has a value
+        foreach(PropertyInfo field in user.GetType().GetProperties()){
+            
+            if(field.GetValue(user) == null)
+                throw new Exception(message: field.Name + " is a mandatory field");
+        }
+        
+        //Establish the values of password and expireDate before calling the repository
+
+        //Set Initial Password
```

Comment:
Yes, I actually tought about doing it, but hesitated because the rest of the classes where the service functions, and another class seemed a lot for that amount of code.

Regardless, after finishing Unit testing, I will separate it into a different method. Thx for the feedback!